### PR TITLE
[YUNIKORN-1252] expose timed out placeholders count in application REST info

### PR DESCRIPTION
**Pull request by steinsgateted**
### What is this PR for?
add timed out placeholders count in application REST info and add some placeholderData tests

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1252

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/webservice/dao/application_info.go [hunk 1]
```diff
@@ -48,4 +48,5 @@ type PlaceholderDAOInfo struct {
 	Count         int64            `json:"count"`
 	MinResource   map[string]int64 `json:"minResource"`
 	Replaced      int64            `json:"replaced"`
+	Timeout       int64            `json:"timeout"`
```
Comment (wilfred-s) [inline pkg/webservice/dao/application_info.go:51]: Missing "d" in the name: Time**d**Out Also capitalise the `O`
json field should be timedout also.

File: pkg/webservice/handlers.go [hunk 2]
```diff
@@ -240,6 +240,7 @@ func getApplicationJSON(app *objects.Application) *dao.ApplicationDAOInfo {
 			Count:         taskGroup.Count,
 			MinResource:   taskGroup.MinResource.DAOMap(),
 			Replaced:      taskGroup.Replaced,
+			Timeout:       taskGroup.Timedout,
```
Comment (wilfred-s) [inline pkg/webservice/handlers.go:243]: Missing "d" in the name: Time**d**Out Also capitalise the `O`

**Review by wilfred-s — CHANGES_REQUESTED** [review]
I do not think we really test if the timeout counter is updated and then correctly exposed.
If you rely on the counter to be correct and that is covered in other tests than OK, however seeing a handler test that covers it would be good.

File: pkg/webservice/dao/application_info.go (hunk 1 above)
Comment (steinsgateted) [reply pkg/webservice/dao/application_info.go]: Done.Thanks.

File: pkg/webservice/handlers.go (hunk 2 above)
Comment (steinsgateted) [reply pkg/webservice/handlers.go]: Done.Thanks.

**Comment by steinsgateted** [comment]
@wilfred-s Thanks for your review.
In `application.go`, add a `SetTimedOutPlaceholder` func for handler test.

**Review by wilfred-s — APPROVED** [review]
LGTM +1

**Pull request closed without merging**